Mathlib/Order/Nucleus.lean

```diff
@@ -0,0 +1,104 @@
+/-
+Copyright (c) 2024 Christian Krause. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chriara Cimino, Christian Krause
+-/
+import Mathlib.Order.CompleteLattice
+import Mathlib.Order.Hom.Lattice
+
+/-!
+# Nucleus
+Locales are the dual concept to Frames.
+A Nucleus is a function between Frames which corresponds to a sublocale.
+https://ncatlab.org/nlab/show/nucleus
+-/
+
+open Order
+
+variable {X : Type*} [CompleteLattice X]
+
+/--
+A Nucleus is a function between Frames which corresponds to a sublocle. It is idempotent,
+increasing and preserves infima.
+-/
+structure Nucleus (X : Type*) [SemilatticeInf X] extends InfHom X X where
+  /-- A Nucleus is idempotent.-/
+  idempotent' (x : X) : toFun (toFun x) ≤ toFun x
+  /-- A Nucleus is increasing.-/
+  increasing' (x : X) : x ≤ toFun x
+
+/--
+A stronger version of Nucleus.idempotent which follows from Nucleus.increasing.
+-/
```

Comment:
I don't think that docstring brings any value since `Nucleus.idempotent'` is an implementation detail
```suggestion
```